pandas/io/packers.py

```diff
@@ -184,6 +184,8 @@ def read(fh):
         if exists:
             with open(path_or_buf, 'rb') as fh:
                 return read(fh)
+        else:
+            return FileNotFoundError('{} not found'.format(path_or_buf))
```

Comment:
although this may simplify testing, i'd be inclined to just let Python raise the system specific exception. 

i.e.

replace
```python
    # see if we have an actual file
    if isinstance(path_or_buf, str):
        try:
            exists = os.path.exists(path_or_buf)
        except (TypeError, ValueError):
            exists = False

        if exists:
            with open(path_or_buf, 'rb') as fh:
                return read(fh)
```

with
```python
    if isinstance(path_or_buf, str):
        with open(path_or_buf, 'rb') as fh:
            return read(fh)
```

and it raises (on windows)..
``` 
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\simon\\OneDrive\\code\\pandas-simonjayhawkins\\pandas\\tests\\io\\data\\does_not_exist.mp'
```

Reply:
Agreed